pkg/action/action_lock_store.go

```diff
@@ -94,20 +94,17 @@ func (a *ActionLockStore) getLockKey(actionItem *proto.ActionItemDTO) (string, e
 // Gets lock key for the pod. For a bare pod, the key is its (first) container name + image name. Otherwise, the key is the formatted string:
 // [parentKind]-[pod.Namespace]/[parentName] with its parent controller.
 func getPodLockKey(pod *api.Pod) (string, error) {
-
-	parentKind, parentName, _, err := podutil.GetPodParentInfo(pod)
+	ownerInfo, err := podutil.GetPodParentInfo(pod)
 	if err != nil {
 		glog.Errorf("Failed to get pod[%s] parent info: %v", util.BuildIdentifier(pod.Namespace, pod.Name), err)
 		return "", err
 	}
-
-	// If the pod is not a bare pod, the key is the parent name.
-	if parentKind != "" { // Not a bare pod
-		return fmt.Sprintf("%v-%v/%v", parentKind, pod.Namespace, parentName), nil
+	if podutil.IsOwnerInfoEmpty(ownerInfo) {
+		// For bare-pod case, the pod name and uid will change after actions applied. So, it's not safe to use as key.
+		// Here, use the (first) container name + the image name as the key to safely lock the pod.
```

Comment:
@irfanurrehman, as you can see, the core part of this logic is not changed, the change here is merely a refactoring of how we get the owner of the pod. 

I believe this logic is introduced in https://github.com/turbonomic/kubeturbo/pull/169. What it is trying to do is if 
 `kubeturbo` receives two actions (e.g., resize and move) for the same bare pod at the same time, it needs to locate the correct pod. 

For example:
action1: move `pod1`
action2: resize `pod1`

After action1, `pod1` -> `pod1-foo`
Then for action2, `kubeturbo` needs to locate `pod1-foo` by looking up the cache, and apply the resize action on it.

I think your concern is right that using the container name and image name as the key may have conflicts, but it is not relevant to this PR.

I wonder if this can be investigated and addressed along with the bare pod move ticket? Again, I don't believe the priority is high, as moving and resizing bare pods these days are really rare.